### PR TITLE
OSD-10193: added some more infos to contribution guide

### PR DESCRIPTION
**Pull request by ramonbutter**
This is a collection of information that can help to contribute and extend configuration-anomaly-detection

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,20 +1,58 @@
...
+CAD is in the end deployed on an OpenShift cluster. Different configurations are set depending on the environment it serves (in, stage, prod). However, when you design the investigation functions, keep in mind how the investigation will be runnable from the local machine, without the need to deploy them first into Kubernetes. This means that required configurations should be injected via env-variables to the CLI and not be fetched by f.e. a config map, and think of the dependent values.
+
+The dependencies so far are:
+
+- AWS
+- Pagerduty
+- OCM
+- Tekton
+  
+For testing an investigation, it may be sufficient to run CAD locally. A JSON blob with an incident ID from PD should be used as an argument. However, make sure you trigger the particular incident in a cluster.
```
Comment (georgettica) [inline CONTRIBUTING.md:37]: Do we have a hard openshift requirement?
Does OLM install seamlessly on k8s aswell and the `subscription` can be worked the same?

---
We do expose a route so maybe we are vendor locking a bit

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,20 +1,58 @@
...
+To test Tekton and the deployment configuration, put CAD on OpenShift behind a Tekton event-listener and use curl to trigger pipeline runs by using example payloads.
+
+## Coding Style
+
+Standard software engineering practices apply here:
+Please use small functions. Add comments when needed. Think about naming. Export reusable functionality. Use a linter. Try to apply a hexagonal architecture pattern. Add unit tests. Ask peers to review your PRs. Follow the [golang coding style](https://go.dev/doc/effective_go).
```
Comment (georgettica) [inline CONTRIBUTING.md:43]: I am not sure we are doing an hexagonal architecture

**Review by georgettica — COMMENTED** [review]
Also a toil card we can do, make the error function also show the lines which have duplicate error messages

File: CONTRIBUTING.md (hunk 2 above)
Comment (ramonbutter) [reply CONTRIBUTING.md:43]: what else? That pattern matched the most to what we did so far

Comment (georgettica) [reply CONTRIBUTING.md:43]: Ok,  good to know 😅
Unintended hexagonal developer

File: CONTRIBUTING.md (hunk 1 above)
Comment (shibumi) [reply CONTRIBUTING.md:37]: Installing OLM in vanilla k8s should work, but routes won't work and you have to use an ingress instead.

Comment (georgettica) [reply CONTRIBUTING.md:37]: I guess we should caveat this at some point :)
Aside of that it's ok

**Comment by georgettica** [comment]
@ramonbutter add description?

File: CONTRIBUTING.md (hunk 1 above)
Comment (ramonbutter) [reply CONTRIBUTING.md]: I mean our CD pipelines are solely designed for OpenShift. However, I don't think it's difficult to make the setup work with k8s. As @shibumi pointed out the route needs to be replaced with ingress from the CRs side. But not too sure whether we actually need this right now in the CONTRIBUTING guide as we are anyway pretty much locked to our setup, meaning AWS STS clusters with 2 hops in the role chain. Cluster alerts propagate to PD. Maybe we can give more hints when there is a maintenance sprint, or?

**Comment by georgettica** [comment]
we can merge this. but I wouldn't say this task is done :/

**Comment by shibumi** [comment]
@georgettica  what is missing?

**Comment by georgettica** [comment]
What we discussed offline, the input from a functional lead on what would it require to become a maintainer

**Comment by rafael-azevedo** [comment]
> 

I will be adding that information in a seperate PR but I think this one is good to go

**Pull request merged**